### PR TITLE
Pull out ScheduledTaskState.IsDeleted , ScheduledTaskState.TenantId, ScheduledTaskState.Task.IsEnabled into Stored Virtual Columns and Index them

### LINKED ISSUE #361: Listing Tasks takes too long
### Describe the bug

There is a table scan happening when  you get a list of tasks by tenant id.

### Steps to reproduce

1. Add lots of tasks
2. Get a list of them by tenant id
3. observe performance issue

### Expected behaviour

Queries should be fast.

### PR DESCRIPTION
**Pull request by ElanHasson**
There are performance issues with getting lists of tasks by tenant id, so we must create virtual columns for the json properties ScheduledTaskState.IsDeleted , ScheduledTaskState.TenantId, ScheduledTaskState.Task.IsEnabled into Stored Virtual Columns and Index them.

Also used pre-compiled models.

Resolves #361 

## Generated SQL

```sql
START TRANSACTION;

ALTER TABLE `OrleansStorage` ADD `IsScheduledTaskDeleted` tinyint(1) AS (CASE WHEN GrainTypeHash = 2108290596 THEN
    CASE WHEN JSON_EXTRACT(PayloadJson, '$.isDeleted') IS NOT NULL THEN
        true
    ELSE
        false
    END
END) NULL;

ALTER TABLE `OrleansStorage` ADD `IsScheduledTaskEnabled` tinyint(1) AS (CASE WHEN GrainTypeHash = 2108290596 THEN
    CASE WHEN JSON_EXTRACT(PayloadJson, '$.task.isEnabled') IS NOT NULL THEN
        true
    ELSE
        false
    END
END) NULL;

ALTER TABLE `OrleansStorage` ADD `ScheduledTaskCreatedAt` DATETIME(6) AS (CASE WHEN GrainTypeHash = 2108290596 AND IsScheduledTaskDeleted = false THEN
        STR_TO_DATE(REPLACE(JSON_UNQUOTE(JSON_EXTRACT(PayloadJson, '$.task.createdAt')), 'Z','+0000'), '%Y-%m-%dT%H:%i:%s.%f+0000')
END) NULL;

ALTER TABLE `OrleansStorage` ADD `TenantId` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci AS (CASE WHEN GrainTypeHash = 2108290596 THEN
    CASE WHEN JSON_UNQUOTE(JSON_EXTRACT(PayloadJson, '$.tenantId')) IS NOT NULL THEN
        JSON_UNQUOTE(JSON_EXTRACT(PayloadJson, '$.tenantId'))
    ELSE
        JSON_UNQUOTE(JSON_EXTRACT(PayloadJson, '$.tenantIdString'))
    END
END) NULL;

CREATE INDEX `IX_OrleansStorage_ScheduledTaskState_TenantId_IsDeletedEnabled` ON `OrleansStorage` (`TenantId`, `IsScheduledTaskDeleted`, `IsScheduledTaskEnabled`);

INSERT INTO `__EFMigrationsHistory` (`MigrationId`, `ProductVersion`)
VALUES ('20230119194644_Add Virtual Columns for ScheduledTaskState TenantId, IsScheduledTaskDeleted, IsScheduledTaskEnabled, CreatedAt and indexes', '7.0.2');

COMMIT;

```

## Example Query:
```sql
 SELECT GrainIdExtensionString 
    FROM OrleansStorage 
    WHERE  
    TenantId = 'c267a84d-4a45-4cbc-8e76-fde23126d3f7'
    AND IsScheduledTaskDeleted = false
        AND GrainTypeHash=2108290596 -- This is a constant.
    ORDER BY ScheduledTaskCreatedAt ASC
    LIMIT 0, 1000;
```

## The Explain:

```json
{
    "query_block": {
        "select_id": 1,
        "cost_info": {
            "query_cost": "0.35"
        },
        "ordering_operation": {
            "using_filesort": true,
            "table": {
                "table_name": "OrleansStorage",
                "access_type": "ref",
                "possible_keys": [
                    "IX_OrleansStorage_ScheduledTaskState_TenantId_IsDeletedEnabled"
                ],
                "key": "IX_OrleansStorage_ScheduledTaskState_TenantId_IsDeletedEnabled",
                "used_key_parts": [
                    "TenantId",
                    "IsScheduledTaskDeleted"
                ],
                "key_length": "770",
                "ref": [
                    "const",
                    "const"
                ],
                "rows_examined_per_scan": 1,
                "rows_produced_per_join": 0,
                "filtered": "10.00",
                "cost_info": {
                    "read_cost": "0.25",
                    "eval_cost": "0.01",
                    "prefix_cost": "0.35",
                    "data_read_per_join": "438"
                },
                "used_columns": [
                    "GrainTypeHash",
                    "GrainIdExtensionString",
                    "PayloadJson",
                    "Id",
                    "IsScheduledTaskDeleted",
                    "ScheduledTaskCreatedAt",
                    "TenantId"
                ],
                "attached_condition": "(`taskscheduler`.`OrleansStorage`.`GrainTypeHash` = 2108290596)"
            }
        }
    }
}
```

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ env:
   mysqlLegacySqlMode: ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
   # Currently no ONLY_FULL_GROUP_BY, see #1167:
   maxConnections: 255
-  "ConnectionStrings__Default": server=127.0.0.1;user id=root;password=Password12!;port=3306;sslmode=Preferred;database=Orleans;
+  "ConnectionStrings__Default": server=127.0.0.1;user id=root;password=Password12!;port=3306;sslmode=Preferred;database=Orleans;DefaultCommandTimeout=10000000;
```
Comment (piyush0101) [inline .github/workflows/build.yml:26]: curious why this change is required?

**Review by piyush0101 — COMMENTED** [review]
lgtm overall but give me some part of tomorrow as well to fully review it

if someone else can review in the meantime, that's even better

**Review by nitishkumar71 — APPROVED** [review]
I am not well versed with .net but most of the work looks migration. So approving it.

File: .github/workflows/build.yml (hunk 1 above)
Comment (ElanHasson) [reply .github/workflows/build.yml:26]: Good call out -- that connection string only is for running the tests of the migrations.

Right now it doesn't matter as there is no data in the DB during the migration. Once functional tests are added, we'll have seeded data in the DB and the migrations will operate on that DB and larger data sets take more time. 

That said, I'll remove it for now and leave it for when it actually becomes a problem.

Comment (ElanHasson) [inline .github/workflows/build.yml:26]:
```suggestion
  "ConnectionStrings__Default": server=127.0.0.1;user id=root;password=Password12!;port=3306;sslmode=Preferred;database=Orleans;
```

**Review by piyush0101 — APPROVED** [review]
